docs/adr/adr_0012.md

```diff
@@ -0,0 +1,129 @@
+
+## Decision
+
+In addition to the existing `ScanType`, `ParseDefinition`, `ScanCompletionHook` and `CascadingRule` the following additional cluster-wide scoped resource should be introduced: `ClusterScanType`, `ClusterParseDefinition`, `ClusterScanCompletionHook` and `ClusterCascadingRule`. The behavior of these is detailed in the following sections.
+
+The CRD `Scan` and `ScheduledScan` don't require `ClusterWide` variants as they are tied directly to the execution of the scan jobs which are tied to a namespace. They are don't provide any service / reusability to other secureCodeBox components (unlike the CRDs listed above).
+
+### `ClusterScanType`
+
+Other then being cluster scoped `ClusterScanType`s are identical to the existing `ScanType` CRD.
+
+When a new scan is started and the operator requires the scan job template it should first look for `ScanType`s in the `Scan`s namespace matching the `Scan.spec.scanType` name configured in the scan. If that is not found the operator should look for `ClusterScanType`s with the same name.
+No matter which was picked the `Job` for the scan always has to be started in the namespace of the `Scan`.
+
+#### Issues with ConfigMap / Secret Mounts:
+
+Some features of normal scanTypes will not function normally with ClusterScanTypes, especially mounting values from ConfigMaps / Secrets as files / environment variables into your containers as the ConfigMaps / Secrets are not existing in every namespace and there is not sensible way for to roll them out with our helm chart to roll them out.
+
+When deploying cluster scans types you have to make sure that either:
+
+1. All referenced ConfigMaps / Secrets exist in all namespaces. This will likely be done by a script / operator generating these configmaps and persisting them into every namespace
+2. Remove all dependencies to ConfigMaps / Secrets from your scanTypes and move these either into the container image of the scanner, or by specifing a initContainer in the ScanType which copies these files over to volume shared with the scanner container. Some more references and discussions on ways to improve this in the future can be found in [ADR-0009: Architecture for pre-populating the file system of scanners"](./adr-0009)
+
+### `ClusterParseDefinition`
+
+Other then being cluster scoped `ClusterParseDefinition`s are identical to the existing `ParseDefinition` CRD.
+
+When a scan job has completed and the results need to be parsed, the operator should first look for `ParseDefinition`s in the `Scan`s namespace matching the `scantype.spec.extractResults.type` / `clusterscantype.spec.extractResults.type` name configured in the ScanType.
+If that is not found the operator should look for `ClusterParseDefinition`s with the same name.
+No matter which was picked the `Job` for the parser always has to be started in the namespace of the `Scan`.
+
```

Comment:
I am wondering if there is even a need for namespace specific ParseDefinitions anymore, once cluster-wide parse definitions exist. Is there really a use case where you would parse the results of the same scanner differently, depending on the namespace?